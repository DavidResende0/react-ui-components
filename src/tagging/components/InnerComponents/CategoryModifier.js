import React from 'react';
import PropTypes from 'prop-types';
import { Col, ControlLabel, FormGroup } from 'patternfly-react';
import { __ } from '../../../global-functions';
import TagSelector from './TagSelector';
import TaggingPropTypes from '../TaggingPropTypes';

const CategoryModifier = ({
  tagCategories,
  onTagCategoryChange,
  selectedTagCategory,
  categoryLabel,
}) => (
  <FormGroup>
    <Col xs={12} md={4} lg={4}>
      <ControlLabel>{categoryLabel}</ControlLabel>
    </Col>
    <Col xs={12} md={8} lg={8}>
      <TagSelector
        tagCategories={tagCategories}
        onTagCategoryChange={onTagCategoryChange}
        selectedOption={selectedTagCategory}
      />
    </Col>
  </FormGroup>
);

CategoryModifier.propTypes = {
  tagCategories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired).isRequired,
  selectedTagCategory: TaggingPropTypes.category,
  onTagCategoryChange: PropTypes.func.isRequired,
  categoryLabel: PropTypes.string,
};

CategoryModifier.defaultProps = {
  categoryLabel: __('Category'),
  selectedTagCategory: {},
};

export default CategoryModifier;
