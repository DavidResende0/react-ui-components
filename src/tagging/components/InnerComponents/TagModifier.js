import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Form } from 'patternfly-react';
import { __ } from '../../../global-functions';

const TagModifier = ({ header, hideHeader, children }) => {
  const h = hideHeader ? null : (
    <Row>
      <Col lg={12}>
        <h2>{header}</h2>
      </Col>
    </Row>);
  return (
    <React.Fragment>
      {h}
      <Form horizontal>{children}</Form>
    </React.Fragment>);
};

TagModifier.propTypes = {
  header: PropTypes.string,
  hideHeader: PropTypes.bool,
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

TagModifier.defaultProps = {
  header: __('Add/Modify tag'),
  hideHeader: false,
};

export default TagModifier;
