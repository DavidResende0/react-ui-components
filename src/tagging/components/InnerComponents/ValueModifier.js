import React from 'react';
import PropTypes from 'prop-types';
import { Col, ControlLabel, FormGroup } from 'patternfly-react';
import ValueSelector from './ValueSelector';
import { __ } from '../../../global-functions';
import TaggingPropTypes from '../TaggingPropTypes';

const ValueModifier = ({
  values,
  onTagValueChange,
  selectedTagValue,
  multiValue,
  valueLabel,
}) => (
  <FormGroup>
    <Col xs={12} md={4} lg={4}>
      <ControlLabel>{valueLabel}</ControlLabel>
    </Col>
    <Col xs={12} md={8} lg={8}>
      <ValueSelector
        values={values}
        onTagValueChange={onTagValueChange}
        selectedOption={selectedTagValue}
        multiValue={multiValue}
      />
    </Col>
  </FormGroup>
);

ValueModifier.propTypes = {
  selectedTagValue: TaggingPropTypes.value,
  onTagValueChange: PropTypes.func.isRequired,
  valueLabel: PropTypes.string,
  multiValue: PropTypes.bool,
  values: PropTypes.arrayOf(TaggingPropTypes.value).isRequired,
};

ValueModifier.defaultProps = {
  valueLabel: __('Value'),
  multiValue: true,
};

export default ValueModifier;
