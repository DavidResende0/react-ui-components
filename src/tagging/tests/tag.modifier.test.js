import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import TagModifier from '../components/tagModifier';
import ValueModifier from '../components/valueModifier';
import CategoryModifier from '../components/categoryModifier';

const tagCategories = [
  { description: 'Name', id: 1 },
  { description: 'Number', id: 2 },
  { description: 'Animal', id: 3 },
  { description: 'Food', id: 4 },
];
const animalValues = [
  { description: 'Duck', id: 31 },
  { description: 'Cat', id: 32 },
  { description: 'Dog', id: 33 },
];

const selectedTagCategory = { description: 'animal', id: 1 };
const selectedTagValue = { description: 'duck', id: 1 };

const onChange = x => x;

describe('Tagging modifier', () => {
  it('match snapshot', () => {
    const tree = mount(
      <TagModifier>
        <CategoryModifier
          selectedTagCategory={selectedTagCategory}
          onTagCategoryChange={onChange}
          tagCategories={tagCategories}
        />
        <ValueModifier
          onTagValueChange={onChange}
          selectedTagValue={selectedTagValue}
          multiValue={false}
          tagValues={animalValues}
        />
      </TagModifier>);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
