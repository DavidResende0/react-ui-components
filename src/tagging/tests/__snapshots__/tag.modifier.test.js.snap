// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tagging modifier match snapshot 1`] = `
<TagModifier
  header="Add/Modify tag"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="row"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={12}
      >
        <div
          className="col-lg-12"
        >
          <h2>
            Add/Modify tag
          </h2>
        </div>
      </Col>
    </div>
  </Row>
  <Form
    bsClass="form"
    componentClass="form"
    horizontal={true}
    inline={false}
  >
    <form
      className="form-horizontal"
    >
      <CategoryModifier
        categoryLabel="Category"
        onTagCategoryChange={[Function]}
        selectedTagCategory={
          Object {
            "description": "animal",
            "id": 1,
          }
        }
        tagCategories={
          Array [
            Object {
              "description": "Name",
              "id": 1,
            },
            Object {
              "description": "Number",
              "id": 2,
            },
            Object {
              "description": "Animal",
              "id": 3,
            },
            Object {
              "description": "Food",
              "id": 4,
            },
          ]
        }
      >
        <FormGroup
          bsClass="form-group"
        >
          <div
            className="form-group"
          >
            <Col
              bsClass="col"
              componentClass="div"
              lg={6}
              md={4}
              xs={12}
            >
              <div
                className="col-lg-6 col-md-4 col-xs-12"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                  >
                    Category
                  </label>
                </ControlLabel>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              lg={6}
              md={8}
              xs={12}
            >
              <div
                className="col-lg-6 col-md-8 col-xs-12"
              >
                <TagSelector
                  infoText="Only a single value can be assigned from these categories"
                  onTagCategoryChange={[Function]}
                  selectedOption={
                    Object {
                      "description": "animal",
                      "id": 1,
                    }
                  }
                  tagCategories={
                    Array [
                      Object {
                        "description": "Name",
                        "id": 1,
                      },
                      Object {
                        "description": "Number",
                        "id": 2,
                      },
                      Object {
                        "description": "Animal",
                        "id": 3,
                      },
                      Object {
                        "description": "Food",
                        "id": 4,
                      },
                    ]
                  }
                >
                  <Select
                    arrowRenderer={[Function]}
                    autosize={true}
                    backspaceRemoves={true}
                    backspaceToRemoveMessage="Press backspace to remove {label}"
                    className="selected-option"
                    clearAllText="Clear all"
                    clearRenderer={[Function]}
                    clearValueText="Clear value"
                    clearable={false}
                    closeOnSelect={true}
                    deleteRemoves={true}
                    delimiter=","
                    disabled={false}
                    escapeClearsValue={true}
                    filterOptions={[Function]}
                    ignoreAccents={true}
                    ignoreCase={true}
                    inputProps={Object {}}
                    isLoading={false}
                    joinValues={false}
                    label="animal"
                    labelKey="label"
                    matchPos="any"
                    matchProp="any"
                    menuBuffer={0}
                    menuRenderer={[Function]}
                    multi={false}
                    name="form-field-name"
                    noResultsText="No results found"
                    onBlurResetsInput={true}
                    onChange={[Function]}
                    onCloseResetsInput={true}
                    onSelectResetsInput={true}
                    openOnClick={true}
                    optionComponent={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Name",
                          "value": 1,
                        },
                        Object {
                          "label": "Number",
                          "value": 2,
                        },
                        Object {
                          "label": "Animal",
                          "value": 3,
                        },
                        Object {
                          "label": "Food",
                          "value": 4,
                        },
                      ]
                    }
                    pageSize={5}
                    placeholder="Select..."
                    removeSelected={true}
                    required={false}
                    rtl={false}
                    scrollMenuIntoView={true}
                    searchable={true}
                    simpleValue={false}
                    tabSelectsValue={true}
                    trimFilter={true}
                    value={1}
                    valueComponent={[Function]}
                    valueKey="value"
                  >
                    <div
                      className="Select selected-option has-value is-searchable Select--single"
                    >
                      <input
                        disabled={false}
                        key="hidden.0"
                        name="form-field-name"
                        type="hidden"
                        value="1"
                      />
                      <div
                        className="Select-control"
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                      >
                        <span
                          className="Select-multi-value-wrapper"
                          id="react-select-2--value"
                        >
                          <Value
                            disabled={false}
                            id="react-select-2--value-item"
                            instancePrefix="react-select-2-"
                            onClick={null}
                            placeholder="Select..."
                            value={
                              Object {
                                "label": "Name",
                                "value": 1,
                              }
                            }
                          >
                            <div
                              className="Select-value"
                            >
                              <span
                                aria-selected="true"
                                className="Select-value-label"
                                id="react-select-2--value-item"
                                role="option"
                              >
                                Name
                              </span>
                            </div>
                          </Value>
                          <AutosizeInput
                            aria-activedescendant="react-select-2--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-owns=""
                            className="Select-input"
                            injectStyles={true}
                            minWidth="5"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            value=""
                          >
                            <div
                              className="Select-input"
                              style={
                                Object {
                                  "display": "inline-block",
                                }
                              }
                            >
                              <input
                                aria-activedescendant="react-select-2--value"
                                aria-expanded="false"
                                aria-haspopup="false"
                                aria-owns=""
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                role="combobox"
                                style={
                                  Object {
                                    "boxSizing": "content-box",
                                    "width": "5px",
                                  }
                                }
                                value=""
                              />
                              <div
                                style={
                                  Object {
                                    "height": 0,
                                    "left": 0,
                                    "overflow": "scroll",
                                    "position": "absolute",
                                    "top": 0,
                                    "visibility": "hidden",
                                    "whiteSpace": "pre",
                                  }
                                }
                              />
                            </div>
                          </AutosizeInput>
                        </span>
                        <span
                          className="Select-arrow-zone"
                          onMouseDown={[Function]}
                        >
                          <span
                            className="Select-arrow"
                            onMouseDown={[Function]}
                          />
                        </span>
                      </div>
                    </div>
                  </Select>
                </TagSelector>
              </div>
            </Col>
          </div>
        </FormGroup>
      </CategoryModifier>
      <ValueModifier
        multiValue={false}
        onTagValueChange={[Function]}
        selectedTagValue={
          Object {
            "description": "duck",
            "id": 1,
          }
        }
        tagValues={
          Array [
            Object {
              "description": "Duck",
              "id": 31,
            },
            Object {
              "description": "Cat",
              "id": 32,
            },
            Object {
              "description": "Dog",
              "id": 33,
            },
          ]
        }
        valueLabel="Value"
      >
        <FormGroup
          bsClass="form-group"
        >
          <div
            className="form-group"
          >
            <Col
              bsClass="col"
              componentClass="div"
              lg={6}
              md={4}
              xs={12}
            >
              <div
                className="col-lg-6 col-md-4 col-xs-12"
              >
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  <label
                    className="control-label"
                  >
                    Value
                  </label>
                </ControlLabel>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              lg={6}
              md={8}
              xs={12}
            >
              <div
                className="col-lg-6 col-md-8 col-xs-12"
              >
                <ValueSelector
                  multiValue={false}
                  onTagValueChange={[Function]}
                  selectedOption={
                    Object {
                      "description": "duck",
                      "id": 1,
                    }
                  }
                  tagValues={
                    Array [
                      Object {
                        "description": "Duck",
                        "id": 31,
                      },
                      Object {
                        "description": "Cat",
                        "id": 32,
                      },
                      Object {
                        "description": "Dog",
                        "id": 33,
                      },
                    ]
                  }
                >
                  <Select
                    arrowRenderer={[Function]}
                    autosize={true}
                    backspaceRemoves={true}
                    backspaceToRemoveMessage="Press backspace to remove {label}"
                    clearAllText="Clear all"
                    clearRenderer={[Function]}
                    clearValueText="Clear value"
                    clearable={false}
                    closeOnSelect={true}
                    deleteRemoves={true}
                    delimiter=","
                    disabled={false}
                    escapeClearsValue={true}
                    filterOptions={[Function]}
                    ignoreAccents={true}
                    ignoreCase={true}
                    inputProps={Object {}}
                    isLoading={false}
                    joinValues={false}
                    label="duck"
                    labelKey="label"
                    matchPos="any"
                    matchProp="any"
                    menuBuffer={0}
                    menuRenderer={[Function]}
                    multi={false}
                    name="form-field-name"
                    noResultsText="No results found"
                    onBlurResetsInput={true}
                    onChange={[Function]}
                    onCloseResetsInput={true}
                    onSelectResetsInput={true}
                    openOnClick={true}
                    optionComponent={[Function]}
                    options={
                      Array [
                        Object {
                          "label": "Duck",
                          "value": 31,
                        },
                        Object {
                          "label": "Cat",
                          "value": 32,
                        },
                        Object {
                          "label": "Dog",
                          "value": 33,
                        },
                      ]
                    }
                    pageSize={5}
                    placeholder="Select..."
                    removeSelected={true}
                    required={false}
                    rtl={false}
                    scrollMenuIntoView={true}
                    searchable={true}
                    simpleValue={false}
                    tabSelectsValue={true}
                    trimFilter={true}
                    value={1}
                    valueComponent={[Function]}
                    valueKey="value"
                  >
                    <div
                      className="Select is-searchable Select--single"
                    >
                      <div
                        className="Select-control"
                        onKeyDown={[Function]}
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                      >
                        <span
                          className="Select-multi-value-wrapper"
                          id="react-select-3--value"
                        >
                          <div
                            className="Select-placeholder"
                          >
                            Select...
                          </div>
                          <AutosizeInput
                            aria-activedescendant="react-select-3--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-owns=""
                            className="Select-input"
                            injectStyles={true}
                            minWidth="5"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            value=""
                          >
                            <div
                              className="Select-input"
                              style={
                                Object {
                                  "display": "inline-block",
                                }
                              }
                            >
                              <input
                                aria-activedescendant="react-select-3--value"
                                aria-expanded="false"
                                aria-haspopup="false"
                                aria-owns=""
                                onBlur={[Function]}
                                onChange={[Function]}
                                onFocus={[Function]}
                                required={false}
                                role="combobox"
                                style={
                                  Object {
                                    "boxSizing": "content-box",
                                    "width": "5px",
                                  }
                                }
                                value=""
                              />
                              <div
                                style={
                                  Object {
                                    "height": 0,
                                    "left": 0,
                                    "overflow": "scroll",
                                    "position": "absolute",
                                    "top": 0,
                                    "visibility": "hidden",
                                    "whiteSpace": "pre",
                                  }
                                }
                              />
                            </div>
                          </AutosizeInput>
                        </span>
                        <span
                          className="Select-arrow-zone"
                          onMouseDown={[Function]}
                        >
                          <span
                            className="Select-arrow"
                            onMouseDown={[Function]}
                          />
                        </span>
                      </div>
                    </div>
                  </Select>
                </ValueSelector>
              </div>
            </Col>
          </div>
        </FormGroup>
      </ValueModifier>
    </form>
  </Form>
</TagModifier>
`;
